cmd/wallet-js-sdk/src/oidc/issuance/openid4ci.js

```diff
@@ -1,212 +1,287 @@
+    }
+  * */
+  // TODO: POST as application/x-www-form-urlencoded not json
+  let token_response = await axios.post(
+    transaction_data.issuer_metadata.token_endpoint,
+    tokenRequest,
+  ).then((resp)=> resp.data);
+  // TODO: poll if response is deferred (if token_response.authorization_pending is true),
+  //  waiting token_response.interval seconds before next request (or 5 seconds if interval missing or less than 5).
+
+  return await get_credential(token_response, transaction_data, client_config, jwt_signer);
+}
+
+async function get_credential(
+  token_response,
+  transaction_data,
+  client_config = {
+    user_did: '',
+  },
+  jwt_signer = async function(iss, aud, iat, c_nonce){return ''},
+) {
+  const jwt = await jwt_signer(transaction_data.client_id, transaction_data.issuer_metadata.issuer, /*time now*/0, token_response.c_nonce)
 
-  // TODO note: credential request seems to handle only one credential type at a time, while the original auth request can request multiple credentials.
-  //  should we:
-  //   A) be able to request multiple credentials in one call, or
-  //   B) make multiple requests, one for each requested credential, or
-  //   C) only request one credential at a time in a single issuance flow
   const credentialRequest = {
     type: transaction_data.credential_type,
-    format: '',
-    did: user_did,
+    // format: '', // TODO should we specify format, or allow issuer to always decide by default?
+    did: client_config.user_did,
     proof: {
       proof_type: 'jwt',
       jwt,
     },
   };
 
-  const credentialResponse = await sendCredentialRequest(credentialRequest)
-  // TODO poll for credential if response is deferred
-
-  await delete_transaction(oidc_state);
+  /*
+  Response format:
+  {
+    format: '',
+    credential: {},
+  }
+  */
+  const credentialResponse = await axios.post(transaction_data.issuer_metadata.credential_endpoint, credentialRequest, {
+    headers: {
+      Authorization: "Bearer " + token_response.access_token,
+    },
+  }).then((resp) => resp.data);
+  // TODO deferred flow implementation deferred
 
   return {format: credentialResponse.format, credential: credentialResponse.credential};
 }
+
+
+/*
+get_issuer_metadata returns issuer metadata from the .well-known/openid-configuration endpoint.
+{
+  issuer: uri,
+  authorization_endpoint: uri + '/auth',
+  token_endpoint: uri + '/token',
+  pushed_authorization_request_endpoint: uri + '/par',
+  require_pushed_authorization_requests: true,
+  credential_endpoint: uri + '/credentials',
+}
+*/
+async function get_issuer_metadata(issuer_uri) {
+  return await axios.get(issuer_uri + '/.well-known/openid-configuration')
+    .then((resp) => resp.data)
+    .catch((e) => {
+      throw new Error('Failed to fetch issuer server metadata:', e)
+    });
+}
+
+function parse_transaction(txn) {
+  return JSON.parse(decode(txn))
+}
+
+function marshal_transaction(txn) {
+  return encode(JSON.stringify(txn), true)
```

Comment:
Should it be `base64` or `base64url` encoded?

Reply:
either is fine, this isn't for a URL